snapcraft/tests/matchers.py

```diff
@@ -23,20 +23,29 @@ class HasArchitecture:
 
     def __init__(self, expected_arch):
         self._expected_arch = expected_arch
-        self._ms = magic.open(magic.NONE)
-        self._ms.load()
 
     def __str__(self):
         return 'HasArchitecture()'
 
     def match(self, file_path):
-        magic = self._ms.file(file_path)
-        # Catch exceptions on splitting the string to provide context
-        # This includes "cannot open `...' (No such file or directory)"
+        cmd = ['file', file_path]
         try:
-            arch = magic.split(',')[1]
+            output = subprocess.check_output(cmd).decode()
+        except subprocess.CalledProcessError as cmd_err:
+            raise ValueError(
+                'Failed to get architecture info: {}', cmd_err) from cmd_err
+
+        try:
+            # output is of the form
+            # ELF 64-bit LSB executable, ARM aarch64, version 1 (SYSV), \
+            # dynamically linked, interpreter /lib/ld-linux-aarch64.so.1, \
+            # for GNU/Linux 3.7.0, \
+            # BuildID[sha1]=58b8495bef7a6c9e5f403879e29dc63ad4ed2d11, \
+            # not stripped
+            arch = output.split(',')[1]
```

Comment:
On amd64 I get:

`<file name>: ELF 64-bit LSB executable, x86-64, version 1 (GNU/Linux), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 2.6.32, BuildID[sha1]=a4b02c379f4482e7cb3148e0d3fc9459a9126cfd, not stripped`

`output.split(',')[1]` is ` x86-64` (notice the space). Might want to strip this.

Reply:
the original implementation is exactly the same, except that we call `file` directly.

This is why we do `expected_arch in arch`. Notice that on arm64 `output.split(',')[1]` is `ARM aarch64`

Reply:
Ah yes!